indico/modules/attachments/templates/_attachments.html

```diff
@@ -8,14 +8,18 @@
 
 {% macro render_attachment(attachment, has_label=false, classes='') %}
     {% set previewable = (attachment.file.is_previewable and not g.static_site) or false %}
+    {% set restricted = attachment.is_self_protected and not attachment.can_access(session.user) %}
+    {% set visible_link = attachment.type.name == 'link' and not restricted %}
     <a class="attachment {{ get_attachment_icon(attachment) }} {% if attachment.is_self_protected %}is-protected{% endif %} {% if has_label %}has-label{% endif %} {%- if previewable %} js-preview-dialog{% endif %} {{ classes }}"
        data-previewable="{{ previewable|tojson }}"
        data-attachment-id="{{ attachment.id }}"
-       href="{{ attachment.download_url }}" target="_blank"
+       href="{{ attachment.link_url if visible_link else attachment.download_url }}" target="_blank"
        title="{{ attachment.description or attachment.title }}
               {%- if attachment.is_self_protected %} ({% trans %}Protected{% endtrans %}){% endif %}">
         {%- if not has_label -%}
-            <span class="title">{{ attachment.title }}</span>
+            <span class="title">
+                {{ attachment.link_url if visible_link else attachment.title }}
```

Comment:
i don't think that's a good idea - the link may be awfully long, and we want to keep the title. people can copy the URL or check the status bar if they want to know where the link goes IMO

Reply:
Alright, I added after testing manually and noticing we already have some pretty nice truncation (could be more user friendly):
![imagem](https://user-images.githubusercontent.com/12183954/99828385-c9547480-2b52-11eb-8ff1-786d63bb16b9.png)
But I'm guessing this steals the purpose of link titles.

Reply:
exactly ;) and it would probably look worse in meetings where the attachments are shown more like buttons